packages/compass-sidebar/src/components/multiple-connections/saved-connections/saved-connection.tsx

```diff
@@ -0,0 +1,264 @@
+    <Icon size={spacing[3]} className={iconStyles} glyph="Favorite" />
+  ) : (
+    <ServerIcon />
+  );
+
+  async function copyConnectionString(connectionString: string) {
+    try {
+      await navigator.clipboard.writeText(connectionString);
+      openToast('copy-to-clipboard', {
+        title: 'Success',
+        description: 'Copied to clipboard.',
+        variant: 'success',
+        timeout: TOAST_TIMEOUT_MS,
+      });
+    } catch (err) {
+      openToast('copy-to-clipboard', {
+        title: 'Error',
+        description:
+          'An error occurred when copying to clipboard. Please try again.',
+        variant: 'warning',
+        timeout: TOAST_TIMEOUT_MS,
+      });
+    }
+  }
+
+  const actions: (ItemAction<Action> | ItemSeparator)[] = [
+    {
+      action: 'edit-connection',
+      label: 'Edit connection',
+      icon: 'Edit',
+    },
+    {
+      action: 'copy-connection-string',
+      label: 'Copy connection string',
+      icon: 'Copy',
+    },
+    {
+      action: 'toggle-favorite',
+      label:
+        connectionInfo.savedConnectionType === 'favorite'
+          ? 'Unfavorite'
+          : 'Favorite',
+      icon: 'Favorite',
+    },
+    {
+      action: 'duplicate-connection',
+      label: 'Duplicate',
+      icon: 'Clone',
+    },
+    { separator: true },
+    {
+      action: 'remove-connection',
+      label: 'Remove',
+      icon: 'Trash',
+      variant: 'destructive',
+    },
+  ];
+
+  const onAction = useCallback(
+    (action: Action) => {
+      switch (action) {
+        case 'edit-connection':
+          return onEditConnection(connectionInfo);
+        case 'copy-connection-string':
+          return void copyConnectionString(
+            maybeProtectConnectionString(
+              connectionInfo.connectionOptions.connectionString
+            )
+          );
+        case 'toggle-favorite':
+          return onToggleFavoriteConnection(connectionInfo);
+        case 'duplicate-connection':
+          return onDuplicateConnection(connectionInfo);
+        case 'remove-connection':
+          return onDeleteConnection(connectionInfo);
+      }
+    },
+    [
+      connectionInfo,
+      onEditConnection,
+      copyConnectionString,
+      maybeProtectConnectionString,
+      onToggleFavoriteConnection,
+      onDuplicateConnection,
+      onDeleteConnection,
+    ]
+  );
+  return (
+    <li
+      {...hoverProps}
+      style={{
+        backgroundColor: connectionColorToHex(connectionInfo.favorite?.color),
+      }}
+      className={savedConnectionStyles}
+    >
+      {icon}{' '}
+      <div className={savedConnectionNameStyles}>
+        {getConnectionTitle(connectionInfo)}
+      </div>
+      <div
+        style={{ visibility: isHovered ? 'visible' : 'hidden' }}
+        className={savedConnectionToolbarStyles}
+      >
+        <IconButton
+          onClick={() => onConnect(connectionInfo)}
+          data-testid="connect-button"
+          aria-label="Connect"
```

Comment:
Maybe worth adding a `title` here too so that it can show up when you hover over the button, I couldn't figure out what this icon does without looking at the source